Kitodo/src/main/java/org/kitodo/production/services/data/MassImportService.java

```diff
@@ -100,10 +85,24 @@ public List<CsvRecord> parseLines(List<String> lines, String separator) {
         List<CsvRecord> records = new LinkedList<>();
         for (String line : lines) {
             List<CsvCell> cells = new LinkedList<>();
-            for (String value : line.split(separator, -1)) {
-                cells.add(new CsvCell(value));
+            if (!Objects.isNull(line) && !line.isBlank()) {
+                StringBuilder currentCell = new StringBuilder();
+                boolean inQuotes = false;
+                for (char c : line.toCharArray()) {
+                    if ((c == '\"' || c == '\'') && !inQuotes) {
+                        inQuotes = true;
+                    } else if (c == '\"' || c == '\'') {
+                        inQuotes = false;
+                    } else if (String.valueOf(c).equals(separator) && !inQuotes) {
+                        cells.add(new CsvCell(currentCell.toString()));
+                        currentCell.setLength(0); // Reset currentCell
+                    } else {
+                        currentCell.append(c);
+                    }
+                }
+                cells.add(new CsvCell(currentCell.toString()));
+                records.add(new CsvRecord(cells));
```

Comment:
we could make that more resilient by not adding a record where a quoted value has not closing quotation. 

```
1233,TEST,Name
1234,'TEST,Name
```
```java
 if (!inQuotes){
                    cells.add(new CsvCell(currentCell.toString()));
                    records.add(new CsvRecord(cells));
                }
```

Reply:
Did you abandon this idea?